tools/benchmark/src/main/java/com/datadog/benchmark/internal/reader/MemoryVitalReader.kt

```diff
@@ -7,16 +7,41 @@
 package com.datadog.benchmark.internal.reader
 
 import java.io.File
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
 
 /**
  * Reads the device's `VmRSS` based on the `/proc/self/status` file.
  * cf. documentation https://man7.org/linux/man-pages/man5/procfs.5.html
  */
 internal class MemoryVitalReader(
-    internal val statusFile: File = STATUS_FILE
+    internal val statusFile: File = STATUS_FILE,
+    private val readerDelay: Long = STATUS_FILE_READ_INTERVAL_MS
 ) : VitalReader {
 
-    override fun readVitalData(): Double? {
+    private var maxMemory: Double = 0.0
+    private val executorService: ScheduledExecutorService = Executors.newSingleThreadScheduledExecutor()
+
+    private val readRunnable: Runnable = Runnable {
+        maxMemory = (readStatusFile() ?: 0.0).coerceAtLeast(maxMemory)
+    }
+
+    override fun readVitalData(): Double {
+        return maxMemory.apply {
```

Comment:
this value is accessible/modifiable from different threads so you will need an AtomicDouble here

Reply:
Good point, since `AtomicDouble` doesn't exist, I use `AtomicLong` to store the data